cmds/exp/tc/main.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2012-20124 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/florianl/go-tc"
+	trafficctl "github.com/u-root/u-root/pkg/tc"
+)
+
+func main() {
+	rtnl, err := tc.Open(&tc.Config{})
+	if err != nil {
+		fmt.Printf("%v\n", err)
+		os.Exit(1)
+	}
+	defer func() {
+		if err := rtnl.Close(); err != nil {
+			fmt.Fprintf(os.Stderr, "could not close rtnetlink socket: %v\n", err)
+		}
+	}()
+
+	tctl := &trafficctl.Trafficctl{Tc: rtnl}
+	if err := run(os.Stdout, os.Args[1:], tctl); err != nil {
+		fmt.Printf("%v\n", err)
+		os.Exit(1)
+	}
+}
+
+func run(stdout io.Writer, args []string, tctl trafficctl.Tctl) error {
+	cursor := 0
+	want := []string{
+		"qdisc",
+		"class",
+		"filter",
+	}
+
+	switch one(args[cursor], want) {
+	case "qdisc":
+		return runQdisc(stdout, args[cursor+1:], tctl)
+	case "class":
+		return runClass(stdout, args[cursor+1:], tctl)
+	case "filter":
+		return runFilter(stdout, args[cursor+1:], tctl)
+	}
+
+	return nil
+}
+
+func one(cmd string, cmds []string) string {
+	var x, n int
+	for i, v := range cmds {
+		if strings.HasPrefix(v, cmd) {
+			n++
+			x = i
+		}
+	}
+	if n == 1 {
+		return cmds[x]
+	}
+	return ""
+}
+
+func runQdisc(stdout io.Writer, args []string, tctl trafficctl.Tctl) error {
```

Comment:
sorry I missed this before, but our general pattern in these kinds of functions is to put the args[]string as the last arg. This makes it more convenient when we have a function that can take args ...string as opposed to args []string

Reply:
No you didnt miss it, I just forgot :sweat_smile: